lib/OpenQA/Script/Client.pm

```diff
@@ -150,6 +151,11 @@ sub run {
         $url->query([%params]) if %params;
     }
 
+    if (defined $params{template}) {
+        my $t = Mojo::URL->new($params{template});
+        $params{template} = Mojo::File->new($t->path)->slurp if ($t->scheme // '') eq 'file';
+    }
+
```

Comment:
Why does this need to be duplicated? How about moving the code into OpenQA::Client into a function and calling that function in both places where needed?

Reply:
See my reply to @Martchus in https://github.com/os-autoinst/openQA/pull/3810#discussion_r599802300

Basically I figured that this needed to be extended in both clients, hence the code duplication as already there is some duplicated code among both clients. I will move it to a function in `OpenQA::Client` as you suggest.

Reply:
Taking a look at `OpenQA::Client`, I don't feel this would be a good place for such a function. WDYT about placing it in `OpenQA::YAML` instead?

Reply:
Maybe just don't expand the old client with the functionality at all.

Reply:
that would be ok as well

Reply:
Understood. I will drop the changes from `openqa-client`, and then explore the `--param-file` suggestion by @perlpunk